src/script/cpp_api/s_player.cpp

```diff
@@ -123,6 +124,20 @@ bool ScriptApiPlayer::on_prejoinplayer(
 	return false;
 }
 
+bool ScriptApiPlayer::can_bypass_userlimit(const std::string &name, const std::string &ip)
+{
+	SCRIPTAPI_PRECHECKHEADER
+
+	// Get core.registered_on_prejoinplayers
+	lua_getglobal(L, "core");
+	lua_getfield(L, -1, "registered_can_bypass_userlimit");
+	lua_pushstring(L, name.c_str());
+	lua_pushstring(L, ip.c_str());
+	runCallbacks(2, RUN_CALLBACKS_MODE_OR);
+	FATAL_ERROR_IF(!lua_isboolean(L, -1), "on_user_limitcheck must return a boolean");
```

Comment:
Wouldn't this also error out if `nil` is returned or some other Lua value that modders would expect to be converted to a boolean by `lua_toboolean`?

Reply:
what do you mean ? returning nothing in the callback ?

Reply:
if nil is return, nil is not a boolean